src/main/java/com/varchenko/MikeHashMap.java

```diff
@@ -0,0 +1,282 @@
+            Node<K, V>[] oldBucket = buckets;
+            buckets = new Node[capacity + (capacity >> 1)];
+            for (Node<K, V> kvNode : oldBucket) {
+                if (kvNode != null) {
+                    put(kvNode.getKey(), kvNode.getValue());
+                }
+            }
+        }
+    }
+
+    private int indexFor(Object k) {
+        if (k != null) {
+            int index = k.hashCode() & capacity - 1;
+            return index >= 0 ? index : -index;
+        }
+        return 0;
+    }
+
+    private Node<K, V> getHead(K key) {
+        return buckets[indexFor(key)];
+    }
+
+    private Node<K, V> find(Object key) {
+        return key != null ? findNotNull((K) key) : findNull();
+    }
+
+    private Node<K, V> findNotNull(K key) {
+        Node<K, V> head = getHead(key);
+        if (head != null) {
+            while (head.getKey() != null) {
+                if (head.getKey().hashCode() == key.hashCode() && head.getKey().equals(key)) {
+                    return head;
+                }
+            }
+        }
+        return null;
+    }
+
+    private Node<K, V> findNull() {
+        Node<K, V> head = buckets[0];
+        if (head != null) {
+            while (head.hasNext()) {
+                head = head.next;
+                if (head.getKey() == null) {
+                    return head;
+                }
+            }
+            return head;
+        }
+        return null;
+    }
+
+    @Override
+    public V get(Object key) {
+        Node<K, V> currentNode = find(key);
+        if (currentNode != null) {
+            return currentNode.getValue();
+        }
+        return null;
+    }
+
+    @Override
+    public V put(K key, V value) {
+        resize();
+        Node<K, V> head = getHead(key);
+        if (head == null) {
+            createHead(key, value);
+        } else {
+            return putEntry(head, key, value);
+        }
+        return null;
+    }
+
+    private V putEntry(Node<K, V> head, K key, V value) {
+        Node<K, V> cursor = head;
+        while (cursor.hasNext()) {
+            cursor = cursor.next;
+            if (key != null) {
+                if (cursor.getKey().hashCode() == key.hashCode() && cursor.getKey().equals(key)) {
+                    cursor.setValue(value);
+                } else {
+                    size++;
+                    modCount++;
+                    cursor.setNext(new Node<>(key, value));
+                }
+            } else {
+                cursor.setValue(value);
+            }
+        }
+        cursor.setNext(new Node<>(key, value));
+        size++;
+        modCount++;
+        return value;
+    }
+
+    private void createHead(K key, V value) {
+        buckets[indexFor(key)] = new Node<>(key, value);
+        size++;
+        modCount++;
+    }
+
+    @Override
+    public V remove(Object key) {
+        Node<K, V> current = find(key);
+        if (current != null) {
+            current.setValue(null);
```

Comment:
Method 'remove' should remove Node entirely. But you keep the Node in the MikeHashMap and set up the value to null. It's incorrect.

Reply:
done